api/v1/api.go

```diff
@@ -210,3 +209,71 @@ func (api *API) respondError(w http.ResponseWriter, apiErr apiError, data interf
 		level.Error(api.logger).Log("msg", "failed to write data to connection", "err", err)
 	}
 }
+
+type metricJson map[string]interface{}
+
+func metricsToInterface(metrics []*dto.Metric, metricsType dto.MetricType) []metricJson {
+
+	jsonMetrics := make([]metricJson, len(metrics))
+
+	for i, m := range metrics {
+		individualMetric := make(metricJson)
```

Comment:
How about:
```Go
metric := encodableMetric{}
```
(Includes name change suggested above.)

`individualMetric` is quite verbose. `metric` says the same (as it is singular).

We have a general style rule in Prometheus to avoid `make`. We only use it if len or cap needs to be specified, or if there is another compelling reason.